src/composables/fungibleTokens.ts

```diff
@@ -0,0 +1,317 @@
+import { watch } from 'vue';
+import camelCaseKeysDeep from 'camelcase-keys-deep';
+import BigNumber from 'bignumber.js';
+import { Encoded, Encoding } from '@aeternity/aepp-sdk';
+import { fetchAllPages, handleUnknownError, toShiftedBigNumber } from '@/utils';
+import type {
+  BigNumberPublic,
+  IDefaultComposableOptions,
+  IToken,
+  ITokenBalanceResponse,
+  ITokenList,
+  ITransaction,
+  TokenPair,
+} from '@/types';
+import { PROTOCOL_AETERNITY, STORAGE_KEYS, TX_DIRECTION } from '@/constants';
+import FungibleTokenFullInterfaceACI from '@/lib/contracts/FungibleTokenFullInterfaceACI.json';
+import AedexV2PairACI from '@/lib/contracts/AedexV2PairACI.json';
+import ZeitTokenACI from '@/lib/contracts/FungibleTokenFullACI.json';
+
+import { aettosToAe, calculateSupplyAmount, categorizeContractCallTxObject } from '@/protocols/aeternity/helpers';
+import { AE_SYMBOL } from '@/protocols/aeternity/config';
+
+import { useAccounts } from './accounts';
+import { useAeSdk } from './aeSdk';
+import { useMiddleware } from './middleware';
+import { useTippingContracts } from './tippingContracts';
+import { createNetworkWatcher } from './networks';
+import { createPollingBasedOnMountedComponents } from './composablesHelpers';
+import { useStorageRef } from './storageRef';
+
+/**
+ * List of all custom tokens available (currently only AE network).
+ * As this list is quite big (hundreds of items) it requires processing optimizations.
+ */
+const availableTokens = useStorageRef<ITokenList>(
+  {},
+  STORAGE_KEYS.fungibleTokenList,
+);
+
+/**
+ * List of tokens (assets) owned by active account with the balance value
+ */
+const tokenBalances = useStorageRef<Record<string, IToken[]>>(
+  {},
+  STORAGE_KEYS.fungibleTokenBalances,
+);
+
+const { onNetworkChange } = createNetworkWatcher();
+const availableTokensPooling = createPollingBasedOnMountedComponents(60000);
+const tokenBalancesPooling = createPollingBasedOnMountedComponents(10000);
+
+export function useFungibleTokens({ store }: IDefaultComposableOptions) {
+  const { getAeSdk } = useAeSdk({ store });
+  const { fetchFromMiddleware } = useMiddleware();
+  const { tippingContractAddresses } = useTippingContracts({ store });
+  const {
+    isLoggedIn,
+    aeAccounts,
+    getLastActiveProtocolAccount,
+  } = useAccounts();
+
+  function getTokenBalance(address?: string) {
```

Comment:
Why the method is called in singular (not in plural), but returning all the balances for the given address?

Reply:
Totally makes sense. I think I've made the same comment to Konrad but he didn't include any fix. I changed the method name to `getAccountTokenBalances`.